### ISSUE TITLE
FlashDevelop versus VSCode compilation time difference

### ISSUE THREAD

**Issue by Adolio** [BowlerHatLLC/vscode-as3mxml#422]
Hello Josh,

I noticed a big difference in terms of compilation time regarding _FlashDevelop_ (~16s) versus _VSCode_ (~34s). Here is the command ran from for each context:

**FlashDevelop** (14140594 bytes written in 16.085 seconds)
```
mxmlc-cli -load-config="C:\AIR\AIR_SDK_32_0/frameworks/air-config.xml" -load-config+=obj\Imago-DesktopConfig.xml -inline=true +configname=air -swf-version=43 -o obj\Imago-Desktop637081209041197666
Starting java as: java -Dsun.io.useCanonCaches=false -Xms32m -Xmx512m -Duser.language=en -Duser.region=US -Djava.util.Arrays.useLegacyMergeSort=true -Dapplication.home="C:\AIR\AIR_SDK_32_0" -jar "C:\AIR\AIR_SDK_32_0\lib\ascsh.jar"
INITIALIZING: ascsh v0.4 by Jeff Ward, simbulus.com
(fcsh)
Starting new compile.
Loading configuration: C:\AIR\AIR_SDK_32_0\frameworks\air-config.xml
Loading configuration: E:\Development\Air\LFI\obj\Imago-DesktopConfig.xml

14140594 bytes written to E:\Development\Air\LFI\obj\Imago-Desktop637081209041197666 in 16.085 seconds
```

**VSCode** (37171990 bytes written in 34.553 seconds)
```
Configuration file: e:\Development\Air\LFI\asconfig.desktop.json
Reading configuration file...
Validating configuration file...
Parsing configuration file...
SDK: c:\AIR\AIR_SDK_32_0
Compiling application...
C:\Program Files (x86)\Java\jre1.8.0_221\bin\java -Xms512M -Xmx1G -Dflexcompiler=c:\AIR\AIR_SDK_32_0 -Dflexlib=c:\AIR\AIR_SDK_32_0\frameworks -jar c:\AIR\AIR_SDK_32_0\lib\mxmlc-cli.jar +flexlib=c:\AIR\AIR_SDK_32_0\frameworks --debug=false +configname=air --output=bin/desktop/Imago-Desktop.swf --library-path+=lib --library-path+=lib/themes --source-path+=src --source-path+=../DragonBonesAS/DragonBones/src
--source-path+=../DragonBonesAS/Starling/Starling_2.x/src --source-path+=../Starling-Extension-Particle-System/src --default-size 800 480 --swf-version=43 --default-frame-rate=60 --define+=CONFIG::release,true --define+=CONFIG::debug,false --define+=CONFIG::isAir,true --define+=CONFIG::isAirMobile,false --define+=CONFIG::isAirDesktop,true --define+=CONFIG::embedSounds,true --define+=CONFIG::embedLevelChunks,true -- src/ch/adolio/Main.as
Loading configuration: c:\AIR\AIR_SDK_32_0\frameworks\air-config.xml

37171990 bytes written to E:\Development\Air\LFI\bin\desktop\Imago-Desktop.swf in 34.553 seconds
```

Apparently the number of bytes written are significantly different.

- Do you have an idea from where could come this difference?
- Is VSCode options better for a final release build?

Thanks in advance for your help.

Best,
Aurélien

**Comment by joshtynjala** [comment]
It's difficult to say what is causing the size difference. It may be that more classes are being included for some reason. Perhaps some extra embedded assets.

You could try using a couple of different compiler options to see what differences there are:

The most useful would probably be [`size-report`](https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/asconfig.json#size-report). The generated file will show how different parts of your app affect the final file size. Use the option in both VSCode and FlashDevelop and compare the two files.

You may also want to try using [`dump-config`](https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/asconfig.json#dump-config). Assuming that your compiler options are slightly different between FlashDevelop and VSCode, this might give you an idea of what you need to add or remove in VSCode's asconfig.json file.

**Comment by devlfm** [comment]
I have this problem too with FlashBuilder, I tried many configurations, and the most close I got was with `-load-externs=bin-debug/report.xml` on modules asconfig, and `-link-report=bin-debug/report.xml` on the main app asconfig. But the true is that I'm not very confident, when I have some spare time I'll try the size-report too. Pls share results @Adolio

**Comment by Adolio** [comment]
Thanks Josh for the investigation tracks! I didn't know about `size-report` & `dump-config`, very useful!

I finally found the time to restart digging into this... and I found my issue :) It was indeed coming from different compiler options linked to embedded assets. In the first case my sounds assets weren't embedded and in the second they were...

Sorry for the inconvenience & thanks again for the help!
